### PR TITLE
DRY SCP Command, add Copy command

### PR DESCRIPTION
**Pull request by caldempsey**
This pull request (PR) adds the Copy command to the Unweave CLI. The approach is to reuse the existing CopySource function but modify it slightly to make it more versatile, which prevents a large diff.

Before merging, we may want to consider adding some quality-of-life (QoL) improvements such as:
-  Introducing specialized wildcards, like using . to specify the current directory and all its contents.
-  Implementing safeguards to ignore SCP to dangerous paths, such as /, /etc, /var, or /user, on the remote or host.
-   Adding an "in progress" indicator for large copies since copying the entire CLI directory can take time.
-   Automatically creating the destination directory if it does not exist on the remote.

Example flow:  

1. go run . copy /home/mmacheerpuppy/foobar sess:moon-tears-brass-met/home/
2. Make changes to foobar
3. go run . copy sess:moon-tears-brass-met/home/foobar /home/mmacheerpuppy/barfoo

### CODE REVIEW

File: cmd/ssh.go [hunk 1]
```diff
@@ -227,7 +231,7 @@ func createTempContextFile(execID string) (*os.File, error) {
 	return tmpFile, nil
 }
 
-func copySourceSCP(srcPath, tmpDstPath, dstPath string, connectionInfo types.ConnectionInfo, privKeyPath string) error {
+func copySourceSCP(fromTo []string, privKeyPath string) error {
```
Comment (caldempsey) [inline cmd/ssh.go]: We can re-use this method for Copy if we just move the host/destination path up to the call site.

File: main.go [hunk 2]
```diff
@@ -144,6 +144,17 @@ func init() {
 	codeCmd.Flags().IntVar(&config.HDD, "hdd", 0, "Amount of hard-disk space to allocate in GB")
 	rootCmd.AddCommand(codeCmd)
 
+	cpCmd := &cobra.Command{
+		Use:     "cp",
+		Short:   "Copy files and folders to or from a remote host",
+		Long:    "Example: unweave cp local/path sess:<exec-name/id>/absolute/path/on/remote",
```
Comment (noorvir) [inline main.go]: Docs need to have a format for the arguments and something the user can understand (in this case, the user does not understand the concept of `exec`:

```go
Use:   "cp <source-path> <destination-path>",
Short: "Copy files and folders to or from a remote host",
Long: wordwrap.String("Copy files and folders to or from a remote host \n\n"+
	"cp <source-path> <destination-path>\n\n"+
	"To copy to a remote session, you must prefix the remote path with the session "+
	"name `sess:<session-name>`. The `sess:` prefix is necessary.\n\n"+
	"The full syntax for this command is:\n\n"+
	"Local to remote:\n"+
	"unweave cp <local-path> sess:<session-name><remote-path>\n\n"+
	"Remote to local:\n"+
	"unweave cp sess:<session-name><remote-path> <local-path> \n\n"+
	"Example: \n"+
	"\tunweave cp /home/data sess:session-name/home/ml-data\n"+
	"\tunweave cp sess:session-name/home/ml-data /home/data\n", ui.MaxOutputLineLength),
```

Comment (caldempsey) [reply main.go]: Great, I updated the Cobra docs with this and added a bit about '.'

File: cmd/copy.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+		if argExecID != "" {
+			targetExec, err = getExecByNameOrID(cmd.Context(), argExecID)
+			if err != nil {
+				return fmt.Errorf("Could not find session by name or ID")
+			}
+		}
+
+		formattedArg, err := formatCopyArgToScpArgs(cmd.Context(), argExecID, targetExec, arg)
+		if err != nil {
+			return err
```
Comment (noorvir) [inline cmd/copy.go]: This should not print the error to the user. This should be a `ui.Errof`. The user shouldn't see go errors

File: cmd/ssh.go [hunk 4]
```diff
@@ -182,7 +182,7 @@ func handleCopySourceDir(isNew bool, e types.Exec, privKey string) error {
 			ui.Errorf("Failed to get active project path. Skipping copying source directory")
 			return fmt.Errorf("failed to get active project path: %v", err)
 		}
-		if err := copySource(e.ID, dir, config.ProjectHostDir(), *e.Connection, privKey); err != nil {
+		if err := copySourceAndUnzip(e.ID, dir, config.ProjectHostDir(), *e.Connection, privKey); err != nil {
 			fmt.Println(err)
```
Comment (noorvir) [inline cmd/ssh.go]: Handle error and remove print

File: cmd/copy.go [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+func getExecIDFromCopyArgs(input string) (string, error) {
+	matches := getSessionIDRegex.FindStringSubmatch(input)
+
+	if len(matches) == 2 {
+		return matches[1], nil
+	} else if input == "sess:" {
+		return "", fmt.Errorf("ExecId not specified")
+	} else {
+		return "", nil
```
Comment (noorvir) [inline cmd/copy.go:103]: This is somewhat confusing. I'd expect this to return an error if the exec wasn't parsed. I see you can some conditional logic above for the case where you handle the empty string case. 

If this works it's fine and we can refactor later. Just hard to follow the logic

**Review by noorvir — CHANGES_REQUESTED** [review]

File: cmd/copy.go (hunk 5 above)
Comment (caldempsey) [reply cmd/copy.go:103]: yeah i see what you mean, i'll try to think about a re-write of this in the next pass, but it works for the for loop

**Review by noorvir — APPROVED** [review]
